### DISCUSSION TITLE
Type error on Controller name property

### DISCUSSION THREAD

**Discussion by sirbully** [react-hook-form/react-hook-form#5458] [Q&A] [answered]
**Describe the bug**
Maybe I just don't know how to use this package properly yet but the `name` property in `Controller` expects every dynamic input field in the form

```
Type 'string' is not assignable to type '"email" | "firstName" | "lastName" | "phone" | "company"'.ts(2322)
```

**To reproduce**
1. Open codesandbox link below
2. Check field arrays in `constants/fields.ts`
3. Check `App.tsx`, line 37 and line 61
4. See type error like below:

```
Type 'string' is not assignable to type '"email" | "firstName" | "lastName" | "phone" | "company"'.ts(2322)
controller.d.ts(18, 5): The expected type comes from property 'name' which is declared here on type
'IntrinsicAttributes & { render: ({ field, fieldState, formState, }: { field: 
ControllerRenderProps<RequestAccountProps, "email" | "firstName" | "lastName" | "phone" | "company">; 
fieldState: ControllerFieldState; formState: FormState<...>; }) => ReactElement<...>; } 
& UseControllerProps<...>'
```

**Codesandbox link (Required)**
https://codesandbox.io/s/modest-heisenberg-hqyqw?file=/src/App.tsx

**Expected behavior**
I expect type string to be allowed, otherwise, I can also just write down all the possible string for the forms :(

**Additional context**
typescript@4.2.4
react-hook-form@7.7.1

**Comment by bluebill1049** [comment, marked as answer]
those `string` type should use this type instead: https://react-hook-form.com/ts#FieldPath

**Reply by sirbully** [reply]
Thank you! Heaven-sent answer, I should've just looked more into the docs ✨

**Reply by bluebill1049** [reply]
all good 👍
